sopel/tools/time.py

```diff
@@ -58,16 +78,23 @@ def validate_timezone(zone):
     If the zone is not valid, :exc:`ValueError` will be raised.
 
     .. versionadded:: 6.0
+
+    .. versionchanged:: 8.0
+
+        If ``zone`` is ``None``, raises a :exc:`ValueError` as if it was an
+        empty string or an invalid timezone instead of returning ``None``.
+
     """
     if zone is None:
-        return None
+        raise ValueError('Invalid time zone.')
 
     zone = '/'.join(reversed(zone.split(', '))).replace(' ', '_')
```

Comment:
`pytz.timezone(None)` also raises `UnknownTimeZoneError`, so what do you think of simplifying this to only define the exception type & message in one location?

```py
if zone:
    zone = '/'.join(reversed(zone.split(', '))).replace(' ', '_')

try:
    tz =  # etc.
```

"Applying suggestions on deleted lines is not supported" is such a cop-out, GH. 😑

Reply:
But sadly... I can't do that, because pytz defines the `timezone` function as using an `str` and does not accept `None` as a valid value.

> sopel/tools/time.py:92: error: Argument 1 to "timezone" has incompatible type "Optional[str]"; expected "str"  [arg-type]

So that won't do!